### PR TITLE
Feature/crm product

### PR DESCRIPTION
**Pull request by San88Francisco**
https://app.clickup.com/t/8693vtj71

### CODE REVIEW

File: package-lock.json
Comment (pqoqubbw) [inline package-lock.json:1]: we don't use npm

File: src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx [hunk 1]
```diff
@@ -7,22 +7,49 @@ import {
...
+import { useNavigate, useParams } from 'react-router-dom'
+import axiosInstance from '../../axios/settings'
+import { Report } from 'notiflix/build/notiflix-report-aio'
+import { AxiosError } from 'axios'
+
+type Props = {
+  product: ProductResponse | undefined
+}
+
+const CrmAddNewProductButton: FC<Props> = () => {
```
Comment (pqoqubbw) [inline src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx]: you have props but do not use props at all

File: src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx [hunk 2]
```diff
@@ -7,22 +7,49 @@ import {
...
+  const statusProduct = useSelector(
+    (state: RootState) => state.editProduct.status
+  )
 
-const CrmAddNewProductButton = () => {
+  const statusMapping: { [key: string]: string } = {
+    Новий: 'new',
+    Активний: 'activated',
+    Архівований: 'archived'
+  }
```
Comment (pqoqubbw) [inline src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx]: use BE enum

File: src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx [hunk 3]
```diff
@@ -37,46 +64,99 @@ const CrmAddNewProductButton = () => {
...
+    internalStatus: string
+  ) => {
+    if (internalStatus) {
+      await axiosInstance.put(
+        `api/product/${product_id}/change_status?pr_status=${internalStatus}`
+      )
+    }
+  }
+
+  const changePrices = async (pricesData: any[]) => {
```
Comment (pqoqubbw) [inline src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx]: do not use `any`

File: src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx [hunk 4]
```diff
@@ -37,46 +64,99 @@ const CrmAddNewProductButton = () => {
...
+          })
+        ).unwrap()
+      } else {
+        const product_id = parsedIndex
+        await changePrices(pricesData)
+        await updateProductStatus(product_id, internalStatus)
+        await changePopular(product_id)
+
+        Report.success('Товар успішно відредаговано', '', 'Добре')
+        navigate(-1)
```
Comment (pqoqubbw) [inline src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx]: ?

File: src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx [hunk 5]
```diff
@@ -37,46 +64,99 @@ const CrmAddNewProductButton = () => {
...
+      }
+    } catch (e) {
+      console.log('✌️error --->', e)
```
Comment (pqoqubbw) [inline src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx]: remove `console.log`

File: src/components/Crm-categories/MuiSelect.tsx [hunk 6]
```diff
@@ -24,60 +24,85 @@ type Props = {
     categoryValue: number,
     selectedCategory: number
   ) => void
+  product?: any
```
Comment (pqoqubbw) [inline src/components/Crm-categories/MuiSelect.tsx]: do not use `any`

File: src/components/Crm-categories/MuiSelect.tsx [hunk 7]
```diff
@@ -24,60 +24,85 @@ type Props = {
...
+
   const dispatch = useDispatch<AppDispatch>()
 
   useEffect(() => {
-    dispatch(addData({ type, value: selectedCategory }))
-  }, [selectedCategory])
+    if (selectedCategory !== '') {
+      dispatch(addData({ type, value: selectedCategory }))
+    }
+  }, [selectedCategory, dispatch, type])
```
Comment (pqoqubbw) [inline src/components/Crm-categories/MuiSelect.tsx]:
```suggestion
  }, [selectedCategory, type])
```

File: src/helpers/labelCheckbox.ts [hunk 8]
```diff
@@ -0,0 +1 @@
+export const label = { inputProps: { 'aria-label': 'Checkbox demo' } }
```
Comment (pqoqubbw) [inline src/helpers/labelCheckbox.ts]: why do you need this?

File: src/pages/crm-add-new-product/CrmAddNewProduct.tsx [hunk 9]
```diff
@@ -1,33 +1,100 @@
...
+    activated: 'Активний',
+    archived: 'Архівований'
+  }
+
+  const statusClasses = {
+    Новий: styles.statusNew,
+    Активний: styles.statusActive,
+    Архівований: styles.statusArchive
+  }
+
```
Comment (pqoqubbw) [inline src/pages/crm-add-new-product/CrmAddNewProduct.tsx]: same here

**Review by pqoqubbw — CHANGES_REQUESTED** [review]

File: src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx (hunk 1 above)
Comment (San88Francisco) [reply src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx]: done

File: src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx (hunk 2 above)
Comment (San88Francisco) [reply src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx]: done

File: package-lock.json
Comment (San88Francisco) [reply package-lock.json:1]: done

File: src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx (hunk 3 above)
Comment (San88Francisco) [reply src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx]: done

File: src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx (hunk 4 above)
Comment (San88Francisco) [reply src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx]: done

File: src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx (hunk 5 above)
Comment (San88Francisco) [reply src/components/Crm-add-new-product-button/CrmAddNewProductButton.tsx]: done

File: src/components/Crm-categories/MuiSelect.tsx (hunk 6 above)
Comment (San88Francisco) [reply src/components/Crm-categories/MuiSelect.tsx]: done

File: src/components/Crm-categories/MuiSelect.tsx (hunk 7 above)
Comment (San88Francisco) [reply src/components/Crm-categories/MuiSelect.tsx]: done

File: src/helpers/labelCheckbox.ts (hunk 8 above)
Comment (San88Francisco) [reply src/helpers/labelCheckbox.ts]: remove

File: src/pages/crm-add-new-product/CrmAddNewProduct.tsx (hunk 9 above)
Comment (San88Francisco) [reply src/pages/crm-add-new-product/CrmAddNewProduct.tsx]: done

File: src/components/Crm-add-new-product-table/CrmViewProductTable.tsx [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+    }
+    dispatch(updateProductData(updatedInfo))
+  }
+
+  useEffect(() => {
+    return () => {
+      dispatch(clearProductData())
+      dispatch(setProductStatus(''))
+    }
+  }, [location.pathname, dispatch])
```
Comment (pqoqubbw) [inline src/components/Crm-add-new-product-table/CrmViewProductTable.tsx]:
```suggestion
  }, [location.pathname])
```

File: src/components/Crm-categories-block/CrmCategoriesBlockView.tsx [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+import { ProductResponse } from '../../types'
+import styles from './CrmCategoriesBlock.module.scss'
+
+type Props = {
+  product: ProductResponse | undefined
+}
+
+export const CrmCategoriesBlockView: FC<Props> = ({ product }) => {
+  if (!product) return null
+  return (
```
Comment (pqoqubbw) [inline src/components/Crm-categories-block/CrmCategoriesBlockView.tsx]:
```suggestion
  if (!product) return null
  
  return (
```

**Review by pqoqubbw — CHANGES_REQUESTED** [review]

File: src/components/Crm-categories-block/CrmCategoriesBlockView.tsx (hunk 11 above)
Comment (San88Francisco) [reply src/components/Crm-categories-block/CrmCategoriesBlockView.tsx]: done

File: src/components/Crm-add-new-product-table/CrmViewProductTable.tsx (hunk 10 above)
Comment (San88Francisco) [reply src/components/Crm-add-new-product-table/CrmViewProductTable.tsx]: done

**Pull request merged**